staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -153,6 +159,10 @@ func (t *envelopeTransformer) TransformToStorage(ctx context.Context, data []byt
 	if err != nil {
 		return nil, fmt.Errorf("failed to encrypt DEK, error: %w", err)
 	}
+	// validate annotations
+	if err := validateAnnotations(resp.Annotations); err != nil {
+		return nil, fmt.Errorf("failed to validate annotations: %w", err)
+	}
```

Comment:
This is handled inside of `.doEncode` now right?  Do we have any concern about potential memory leaks caused by the DEK cache growing on the failure case?  I am guessing "no" because your cluster just won't work because KMS will be broken?

Reply:
that's right! In case of error, KMS will be broken.